### ISSUE TITLE
[Fabric 1.19] Starlight cause losses of world data 

### ISSUE THREAD

**Issue by JereW333** [Spottedleaf/Starlight#150]
Context : I tried to switch from a 1.18.2 Paper server to a 1.19 Fabric server. I had a fully functional fabric server where I tested the mods used. When I transferred my world data from one to another, only the entities was correctly placed in the world, none of the players modifications were correct.

How I found the bug ? : I create a new server from scratch and then add the mods I want one at a time to test their impact. It was when Starlight was added that I lost the block modifications data.

Mods used : 
- FabricProxy-Lite-2.0.0
- cloth-config-7.0.69-fabric
- ferritecore-5.0
- lazydfu-0.1.3
- simplebackup-1.3.2
- voicechat-fabric-1.19-2.2.44
- XaerosWorldMap_1.23.2_Fabric_1.19
- fabric-api-0.55.3+1.19
- krypton-0.2.0
- lithium-fabric-mc1.19-0.8.0
- starlight-1.0.2+fabric.89b8d9f

**Comment by Spottedleaf** [comment]
I need logs and steps to how you transferred the world, what dimensions were lost, etc 
Bukkit and vanilla do not have the same layout format, so it's possible if transferred incorrectly you will "lose" end/nether dimensions

**Comment by JereW333** [comment]
I cannot really provide logs because there is no error showing inside the console. But my world data are incorrect due to the lack of blocks placed by players. It was as the world was again generated from seed : in place of a beautiful city, there was the originals mountains with all entities we placed suffocating in walls (villagers, wolfs, etc...).
The dimension were data were lost is The Overworld.
And I knew about the differences between end/nether in the type of servers. I did the correct manipulation, as explain in the paper documentation (chapter on the switch between different types of servers)

**Comment by Spottedleaf** [comment]
Unfortunately I can produce this locally, but I can't do anything about it. You're using an outdated build of starlight, which is the cause. 
Why has it happened? I don't know. I am now investigating, as there were no significant code changes in the light data handling and the lack of errors is worrying.

**Comment by JereW333** [comment]
Oh I didn't know it was an outdated build, I downloaded the last one at the time on modrinth. 
But now that I checked, yep I was using an outdated one... I will try with the latest version.

**Comment by JereW333** [comment]
Ok so it works with the last version. Thank you for your help and good luck finding the source of this bug !

**Comment by Spottedleaf** [comment]
It appears the problem was because the kind of exceptions starlight catches wasn't the one that blew up when loading your old world data. In fact, it should have (and will) blow up on 1.19 regardless of what world data you put in there. Unfortunately for me, when I tested 1.18 starlight on 1.19 I did in a dev environment - where the exception wouldn't occur. So I thought it would have possibly just resulted in light data loss. Specifically, the problem was that a method mapping had changed during the update (big surprise) and the old starlight code was trying to call a method that didn't exist.
Unfortunately for all of us, the vanilla chunk system is **_awful_** and will not log this error.
